src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -374,6 +375,13 @@ namespace cryptonote
     m_core.set_target_blockchain_height((hshd.current_height));
     }
     MINFO(context << "Remote blockchain height: " << hshd.current_height << ", id: " << hshd.top_id);
+
+    if (m_no_sync)
+    {
+      context.m_state = cryptonote_connection_context::state_normal;
+      return true;
+    }
```

Comment:
I'm not 100% sure this will be enough, but I suppose we can fix it if/when it pops up.